pkg/appgw/health_probes_test.go

```diff
@@ -0,0 +1,116 @@
+// -------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// --------------------------------------------------------------------------------------------
+
+package appgw
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/extensions/v1beta1"
+)
+
+func TestHealthProbes(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test setting up App Gateway health probes")
+}
+
+var _ = Describe("configure App Gateway health probes", func() {
+	port1, port2, port3, port4 := makeServicePorts()
+
+	Context("looking at TLS specs", func() {
+		cb := makeConfigBuilderTestFixture(nil)
+
+		endpoints := makeEndpoints()
+		_ = cb.k8sContext.Caches.Endpoints.Add(endpoints)
+
+		service := makeService(port1, port2, port3, port4)
+		_ = cb.k8sContext.Caches.Service.Add(service)
+
+		pod := makePod(testFixturesServiceName, testFixturesNamespace, testFixturesContainerName, testFixturesContainerPort)
+		_ = cb.k8sContext.Caches.Pods.Add(pod)
+
+		ingress := makeIngressTestFixture()
+		ingressList := []*v1beta1.Ingress{
+			&ingress,
+		}
+
+		// !! Action !!
+		_, _ = cb.HealthProbesCollection(ingressList)
+		actual := cb.appGwConfig.Probes
+
+		// We expect our health probe configurator to have arrived at this final setup
+		expected := []network.ApplicationGatewayProbe{
+			{
```

Comment:
Resolving the aobe comments will lead to changes in the list of probes generated.

Reply:
👍